packages/admin-ui/src/Tabs/Tabs.tsx

```diff
@@ -0,0 +1,148 @@
+                lg: "text-lg px-4 py-2",
+                xl: "text-xl px-6 py-3"
+            }
+        },
+        defaultVariants: {
+            size: "md"
+        }
+    }
+);
+
+type TabsTriggerProps = Omit<TabsPrimitive.TabsTriggerProps, "children"> &
+    VariantProps<typeof tabsTriggerVariants> & {
+        text: React.ReactNode;
+        icon?: React.ReactNode;
+    };
+
+const TabsTriggerBase = React.forwardRef<
+    React.ElementRef<typeof TabsPrimitive.Trigger>,
+    TabsTriggerProps
+>(({ className, size, icon, text, ...props }, ref) => (
+    <TabsPrimitive.Trigger
+        ref={ref}
+        className={cn(tabsTriggerVariants({ size }), className)}
+        {...props}
+    >
+        {text} {icon}
+    </TabsPrimitive.Trigger>
+));
+TabsTriggerBase.displayName = TabsPrimitive.Trigger.displayName;
+
+const TabsTrigger = makeDecoratable("TabsTrigger", TabsTriggerBase);
+
+/**
+ * Tabs content
+ */
+const tabsContentVariants = cva(
+    "ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 bg-muted",
+    {
+        variants: {
+            size: {
+                sm: "text-xs p-2",
+                md: "text-sm p-4",
+                lg: "text-lg p-6",
+                xl: "text-xl p-8"
+            }
+        },
+        defaultVariants: {
+            size: "md"
+        }
+    }
+);
+
+type TabsContentProps = Omit<TabsPrimitive.TabsContentProps, "children"> &
+    VariantProps<typeof tabsContentVariants> & {
+        text: React.ReactNode;
+    };
+
+const TabsContentBase = React.forwardRef<
+    React.ElementRef<typeof TabsPrimitive.Content>,
+    TabsContentProps
+>(({ className, size, text, ...props }, ref) => (
+    <TabsPrimitive.Content
+        ref={ref}
+        className={cn(tabsContentVariants({ size }), className)}
+        {...props}
+    >
+        {text}
+    </TabsPrimitive.Content>
+));
+TabsContentBase.displayName = TabsPrimitive.Content.displayName;
+
+const TabsContent = makeDecoratable("TabsContent", TabsContentBase);
+
+/**
+ * Tabs
+ */
+interface TabsProps extends TabsPrimitive.TabsProps {
+    triggers: React.ReactElement<TabsTriggerProps>[];
+    contents: React.ReactElement<TabsContentProps>[];
+    size?: "sm" | "md" | "lg" | "xl";
+}
+
+const TabsBase = React.forwardRef<React.ElementRef<typeof TabsPrimitive.Root>, TabsProps>(
+    ({ defaultValue: baseDefaultValue, size = "md", triggers, contents, ...props }, ref) => {
+        const defaultValue = baseDefaultValue || triggers[0].props.value;
+
+        return (
+            <TabsRoot ref={ref} defaultValue={defaultValue} {...props}>
+                <TabsList>
+                    {triggers.map(trigger => (
+                        <React.Fragment key={`tab-trigger-${trigger.key}`}>
+                            {React.cloneElement(trigger, { size })}
+                        </React.Fragment>
+                    ))}
+                </TabsList>
```

Comment:
Nit pick, but could be simplifed. Same could be done for `contents.map` code below.

```suggestion
                <TabsList>
                    {triggers.map(trigger =>
                        React.cloneElement(trigger, { size, key: `tab-trigger-${trigger.key}` })
                    )}
                </TabsList>
```